libs/curseforge.ts

```diff
@@ -107,7 +109,7 @@ export class Mod
 		}
 	}
 
-	private static readonly ModLUT: StringMap<Mod|null> = {};
+	private static readonly ModLUT: { [_: string]: Mod|null } = {};
 	public static async fromID(id: string, versions: string[]) : Promise<Mod|null>
 	{
 		if(this.ModLUT[id] === undefined)
```

Comment:
* Circular dependency resolution doesn't really belong in the "constructor" for `Mod`s.
* Neither does `versions`. What if two parts of code want to fetch different version sets?

Reply:
Removed circular dependency resolution and version management from the constructor. It only accepts the ID and existing urls now.